src/grt/src/RepairAntennas.cpp

```diff
@@ -94,7 +97,10 @@ bool RepairAntennas::checkAntennaViolations(
 
   makeNetWires(routing, nets_to_repair, max_routing_layer);
   arc_->initAntennaRules();
-  for (odb::dbNet* db_net : nets_to_repair) {
+  omp_set_num_threads(num_threads);
+#pragma omp parallel for schedule(dynamic)
+  for (auto it = nets_to_repair.begin(); it != nets_to_repair.end(); ++it) {
+    odb::dbNet* db_net = *it;
     checkNetViolations(db_net, diode_mterm, ratio_margin);
   }
```

Comment:
@maliberty I added parallelization here, but I'm facing some strange non-determinism in the results. Testing with sky130hd/chameleon, the number of antenna violations reported varies slightly across different runs.

I've made additional changes in the antenna_checker to prevent non-determinism, but now I don't see any other possible sources for it. The only thing I can think of is the access to the antenna_violations_ map that is made in parallel. However, we access it using the dbNet pointer as key, and we should not have duplicated nets in this iteration. Do you have some insights on why this is happening?

Reply:
Is this map pre-populated (all the keys are present)?  If not you are doing an insertion which is not safe.

Reply:
Oh, ok. This is the problem then. I will initialize the map with the keys before going to the parallel part.